ducktape/tests/loader.py

```diff
@@ -177,15 +200,12 @@ def import_modules(self, file_list):
                     # an ImportError or b) the module is valid, but there was
                     # some other error when parsing/executing the module, which
                     # can result in a variety of errors (e.g. IndentationError,
-                    # TypeError, etc). Because of the way we are searching for
-                    # valid modules in this loop, where we expect some of the
-                    # module names we construct to fail to import, we ignore
-                    # ImportError here, but log everything else as an
-                    # error. This does mean we fail to log a subset of import
-                    # errors that we should: if the module is valid but itself
-                    # triggers an ImportError (e.g. typo in an import line),
-                    # that error will be swallowed.
-                    if not isinstance(e, ImportError):
+                    # TypeError, etc)
+                    #
+                    # Log all errors except those caused by modules which don't exist
+
+                    module_does_not_exist = isinstance(e, ImportError) and re.search("No module named", str(e))
```

Comment:
omg, this is a wonderful change! Tracking down `ImportError`s due to bugs in code rather than missing modules was pretty painful before.